service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/configurations/ConfigurationsClient.java

```diff
@@ -341,7 +341,7 @@ public void applyConfigurationContentOnDevice(String deviceId, ConfigurationCont
 
         URL url = IotHubConnectionString.getUrlApplyConfigurationContent(this.hostName, deviceId);
 
-        HttpRequest request = createRequest(url, HttpMethod.POST, content.toConfigurationContentParser().toJson().getBytes(StandardCharsets.UTF_8));
```

Comment:
toJson() always returned empty maps for moduleContent/modulesContent/deviceContent, but toJsonElement() works as expected and has unit test coverage